### PR TITLE
feat: extend functionality of CreateDatabaseParams

### PR DESCRIPTION
**Pull request by richardshiue**
* allow creating more than one database views when using CreateDatabaseParams
* get rid of DatabaseSerde in favor of DatabaseData, and include all the views inside it

### CODE REVIEW

File: collab-database/src/user/user_db.rs [hunk 1]
```diff
@@ -353,19 +349,18 @@ impl WorkspaceDatabase {
...
     Ok(database)
   }
 
-  /// Duplicate the database with the given view id.
-  pub async fn get_database_duplicated_data(
-    &self,
-    view_id: &str,
-  ) -> Result<DatabaseData, DatabaseError> {
+  /// Duplicate the database with the view_id of any view in the database
+  pub async fn get_all_database_data(&self, view_id: &str) -> Result<DatabaseData, DatabaseError> {
```
Comment (appflowy) [inline collab-database/src/user/user_db.rs:361]: how about rename to `get_database_data` ?

File: collab-database/src/database.rs [hunk 2]
```diff
@@ -1157,30 +1197,22 @@ impl Database {
     })
   }
 
-  pub fn duplicate_database(&self) -> DatabaseData {
-    let inline_view_id = self.get_inline_view_id();
+  pub fn get_all_database_data(&self) -> DatabaseData {
```
Comment (appflowy) [inline collab-database/src/database.rs:1200]: to_create_database_params is kind of vague. Only need to regenerate the ids when duplicating a database. Also DatabaseData should not contains biz logic. So i prefer using `duplicate_database`

File: collab-database/src/database.rs [hunk 3]
```diff
@@ -1301,15 +1326,95 @@ pub fn timestamp() -> i64 {
...
-  pub rows: Vec<CreateRowParams>,
+  pub rows: Vec<Row>,
 }
 
 impl DatabaseData {
+  /// Converts DatabaseData to CreateDatabaseParams. If `regenerate` is true,
+  /// the timestamps, row_ids and view_ids will all be regenerated. This is used
+  /// when duplicating a database. If false, these fields remain the same.
+  /// This is used in an importing scenario.
+  pub fn to_create_database_params(self, regenerate: bool) -> CreateDatabaseParams {
```
Comment (appflowy) [inline collab-database/src/database.rs:1344]: DatabaseData should not contain biz logic

File: collab-database/tests/user_test/async_test/script.rs [hunk 4]
```diff
@@ -156,21 +146,62 @@ pub async fn run_script(
...
     } => {
       let w_database =
         workspace_database_with_db(1, Arc::downgrade(&db), Some(config.clone())).await;
       let database = w_database.get_database(&database_id).await.unwrap();
-      let actual = database.lock().to_json_value();
-      assert_json_diff::assert_json_include!(actual: actual, expected: expected);
+      let database_data = database.lock().get_all_database_data();
+      let view = database.lock().get_view("v1").unwrap();
+
+      assert_eq!(
```
Comment (appflowy) [inline collab-database/tests/user_test/async_test/script.rs:159]: using assert_eq! will make the tests hard to maintain. user `assert_json_include` is better

File: collab-database/src/views/view.rs [hunk 5]
```diff
@@ -139,61 +159,64 @@ impl CreateViewParamsValidator {
...
-        groups: self.groups,
-        sorts: self.sorts,
-        field_settings: self.field_settings,
-        deps_fields: vec![],
-        deps_field_setting: vec![],
-      },
-    )
-  }
-}
+  pub fn from_database_data(data: DatabaseData) -> Self {
```
Comment (appflowy) [inline collab-database/src/views/view.rs:169]: how about adding documentation for this function?

**Comment by appflowy** [comment]
LGTM

File: collab-database/src/workspace_database/manager.rs [hunk 6]
```diff
@@ -373,19 +369,18 @@ impl WorkspaceDatabase {
...
+
     Ok(database)
   }
 
-  /// Duplicate the database with the given view id.
-  pub async fn get_database_duplicated_data(
-    &self,
-    view_id: &str,
-  ) -> Result<DatabaseData, DatabaseError> {
+  /// Duplicate the database with the view_id of any view in the database
```
Comment (appflowy) [inline collab-database/src/workspace_database/manager.rs:380]: update these documentations

File: collab-database/tests/database_test/view_test.rs [hunk 7]
```diff
@@ -204,33 +189,42 @@ async fn delete_inline_database_view_test() {
...
+  assert_eq!(duplicated_view.name, format!("{}-copy", view.name));
+  assert_ne!(view.id, duplicated_view.id);
+  // modified and created time should also be different but the test completes within one second.
 }
 
 #[tokio::test]
 async fn duplicate_database_data_serde_test() {
   let database_test = create_database_with_default_data(1, "1").await;
-  let duplicated_database = database_test.duplicate_database();
+  let duplicated_database = database_test.get_database_data();
```
Comment (appflowy) [inline collab-database/tests/database_test/view_test.rs:227]: get_database_data is not duplicate the data

Comment (richardshiue) [reply collab-database/tests/database_test/view_test.rs]: I rename the test and the two variables.

File: collab-database/src/workspace_database/manager.rs (hunk 6 above)
Comment (richardshiue) [reply collab-database/src/workspace_database/manager.rs]: Done

**Pull request merged**